node/network/protocol/src/peer_set.rs

```diff
@@ -102,50 +111,35 @@ impl PeerSet {
 		}
 	}
 
-	/// Get the default protocol version for this peer set.
-	pub const fn get_default_version(self) -> ProtocolVersion {
-		match self {
-			PeerSet::Validation => DEFAULT_VALIDATION_PROTOCOL_VERSION,
-			PeerSet::Collation => DEFAULT_COLLATION_PROTOCOL_VERSION,
-		}
+	/// Get the main protocol version for this peer set.
+	///
+	/// Networking layer relies on `get_main_version()` being the version
+	/// of the main protocol name reported by [`PeerSetProtocolNames::get_main_name()`].
+	pub const fn get_main_version(self) -> ProtocolVersion {
+		MAIN_PROTOCOL_VERSION
 	}
 
-	/// Get the default protocol name as a static str.
-	pub const fn get_default_protocol_name(self) -> &'static str {
-		match self {
-			PeerSet::Validation => VALIDATION_PROTOCOL_V1,
-			PeerSet::Collation => COLLATION_PROTOCOL_V1,
-		}
+	/// Get the max notification size for this peer set.
+	pub fn get_max_notification_size(self, _: IsAuthority) -> u64 {
+		MAX_NOTIFICATION_SIZE
 	}
 
-	/// Get the protocol name associated with each peer set
-	/// and the given version, if any, as static str.
-	pub const fn get_protocol_name_static(self, version: ProtocolVersion) -> Option<&'static str> {
-		match (self, version) {
-			(PeerSet::Validation, 1) => Some(VALIDATION_PROTOCOL_V1),
-			(PeerSet::Collation, 1) => Some(COLLATION_PROTOCOL_V1),
-			_ => None,
+	/// Get the peer set label for metrics reporting.
+	pub fn get_label(self) -> &'static str {
+		match self {
+			PeerSet::Validation => "validation",
+			PeerSet::Collation => "collation",
 		}
 	}
 
-	/// Get the protocol name associated with each peer set as understood by Substrate.
-	pub fn into_default_protocol_name(self) -> Cow<'static, str> {
-		self.get_default_protocol_name().into()
-	}
-
-	/// Convert a peer set and the given version into a protocol name, if any,
-	/// as understood by Substrate.
-	pub fn into_protocol_name(self, version: ProtocolVersion) -> Option<Cow<'static, str>> {
-		self.get_protocol_name_static(version).map(|n| n.into())
-	}
-
-	/// Try parsing a protocol name into a peer set and protocol version.
+	/// Get the protocol label for metrics reporting.
 	///
-	/// This only succeeds on supported versions.
-	pub fn try_from_protocol_name(name: &Cow<'static, str>) -> Option<(PeerSet, ProtocolVersion)> {
-		match name {
-			n if n == VALIDATION_PROTOCOL_V1 => Some((PeerSet::Validation, 1)),
-			n if n == COLLATION_PROTOCOL_V1 => Some((PeerSet::Collation, 1)),
+	/// Unfortunately, labels must be static strings, so we must manually cover them
+	/// for all protocol versions here.
+	pub fn get_protocol_label(self, version: ProtocolVersion) -> Option<&'static str> {
```

Comment:
I'm not convinced that we need this. The metrics should be okay to only work with the "normal" label. Especially as we already use the label for one metric. Not sure why we need two different versions.

WDYT @eskimor

Reply:
I don't really mind. Seeing how many connections are on each version could be interesting though.

Reply:
I kept these version-differentiated labels for now to be able to see how many connections are on each version.